compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -2309,6 +2309,12 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
         let fn_ty = tcx.mk_fn_sig(input_tys, output_ty, decl.c_variadic, safety, abi);
         let bare_fn_ty = ty::Binder::bind_with_vars(fn_ty, bound_vars);
 
+        if let hir::Node::Ty(hir::Ty { kind: hir::TyKind::BareFn(bare_fn_ty), span, .. }) =
+            tcx.hir_node(hir_id)
+        {
+            crate::check::check_abi(tcx, hir_id, *span, bare_fn_ty.abi);
+        }
+
```

Comment:
We perform this check here instead of here (where its performed for functions):
https://github.com/rust-lang/rust/blob/9bad7ba324099d124c77c5b06aebf68e11763f7b/compiler/rustc_hir_typeck/src/lib.rs#L158 

Or here (where its performed for extern blocks):
https://github.com/rust-lang/rust/blob/9bad7ba324099d124c77c5b06aebf68e11763f7b/compiler/rustc_hir_analysis/src/check/check.rs#L756

Since those two places seem not to be called for function pointers.